### PR TITLE
[14.0][server_environment_data_encryption] Avoid raising error when config is missing

### PR DESCRIPTION
**Pull request by florian-dacosta**
Instead we fallback on standard server_environment behavior and display a warning to the user.

Aims to solve https://github.com/OCA/server-env/issues/110

reviews welcome @sebastienbeau @simahawk @ivantodorovich

### CODE REVIEW

**Review by ivantodorovich — APPROVED** [review]

**Review by simahawk — COMMENTED** [review]
LG. Although, it's worth to add a line in the README :wink:

**Comment by florian-dacosta** [comment]
> LG. Although, it's worth to add a line in the README wink

Not sure what to add.
"If the minimal configuration is not done, the module has no effect" but is seems pretty obvious no?

**Comment by simahawk** [comment]
You are right. We should probably log a warning instead. What do you think?

**Comment by florian-dacosta** [comment]
@simahawk 
I've added the warning

File: server_environment_data_encryption/models/server_env_mixin.py [hunk 1]
```diff
@@ -15,11 +15,28 @@
...
+        env = self.env["encrypted.data"]._retrieve_env()
+        key_name = "encryption_key_%s" % env
+        key_str = config.get(key_name)
+        key_exists = key_str and True or False
+        if not key_exists:
+            logging.warning(
+                "The minimal configuration is missing. You need at least to add an "
+                "encryption key for the current environment  : {}. While the "
+                "configuration is missing, the module has no effect".format(env)
+            )
```
Comment (simahawk) [inline server_environment_data_encryption/models/server_env_mixin.py]:
```suggestion
            logging.warning(
                "The minimal configuration is missing. You need at least to add an "
                "encryption key for the current environment: %s. While the "
                "configuration is missing, the module has no effect", env
            )
```

**Review by simahawk — COMMENTED** [review]
can you please rewrite the commit msg?

```
s_environment_data_encryption: no error when config is missing

Fallback to normal server env behavior and log a warning.
```

**Comment by florian-dacosta** [comment]
@simahawk Done

**Comment by simahawk** [comment]
works on runboat:
```
2022-05-10 08:25:27,951 77 WARNING b029161ab-bf0c-4bbb-ad91-211e2b9c46d4 root: The minimal configuration is missing. You need at least to add an encryption key for the current environment  : test. While the configuration is missing, the module has no effect 
```
/ocabot merge patch

Thanks!

**Pull request merged**